SourceCode/Freight/ftt_fr_sales_and_mandate.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+import copy
+green_indices=[12, 13, 18, 19]  # Indices for green technologies
+MANDATE_START_YEAR = 2025
+N_YEARS = 11
+MANDATE_END_YEAR = MANDATE_START_YEAR + N_YEARS
+
+def calculate_mandate_share(year):
+    """Calculate the mandate share based on the year."""
+    if year < MANDATE_START_YEAR:
+        return 0.0
+    elif year >= MANDATE_END_YEAR:
+        return 0.0
+    else:
+        # Linear increase from 0 to 1 between start and end years
+        return (year - MANDATE_START_YEAR) / (MANDATE_END_YEAR - MANDATE_START_YEAR)
+
+def get_enhanced_sales(cap, cap_dt, cap_lag, shares, shares_dt, sales_or_investment_in,
+                       timescales, dt, EV_truck_mandate, year):
+    # Step 1: Calculate basic components
+    cap_growth = cap[:, :, 0] - cap_lag[:, :, 0]
+    share_growth_dt = shares[:, :, 0] - shares_dt[:, :, 0]
+    cap_growth_dt = cap[:, :, 0] - cap_dt[:, :, 0]
+    base_eol = cap_dt[:, :, 0] * dt / timescales
+    share_depreciation = shares_dt[:, :, 0] * dt / timescales
+
+    # Step 2: Initialize sales array
+    sales_dt = np.zeros(sales_or_investment_in.shape)
+
+    # Step 3: Compute initial sales
+    for r in range(cap.shape[0]):
+        for tech in range(cap.shape[1]):
+            if cap_growth_dt[r, tech] > 0:
+                sales_dt[r, tech, 0] = cap_growth_dt[r, tech] + base_eol[r, tech]
+            elif -share_depreciation[r, tech] < share_growth_dt[r, tech] < 0:
+                replacement_sales = (share_growth_dt[r, tech] + share_depreciation[r, tech]) * cap_lag[r, tech, 0]
+                sales_dt[r, tech, 0] = max(0, replacement_sales)
+            else:
+                sales_dt[r, tech, 0] = base_eol[r, tech]
+
+    # Step 4: Apply mandate adjustments with global shares and strict enforcement
+    if EV_truck_mandate[0,0,0] == 1 :
+        mandate_share = calculate_mandate_share(year)
+        for r in range(cap.shape[0]):
+            total_sales = np.sum(sales_dt[r, :, 0])
+            if total_sales > 0:
+                current_green = np.sum(sales_dt[r, green_indices, 0])
+                current_share = current_green / total_sales
+                
+                if current_share < mandate_share:
+                    target_green = total_sales * mandate_share
+                    
+                    # Keep existing global distribution logic
```

Comment:
Change the comment to: "#If green tech present in region, scale with region distribution". Only later are you taking the global values.